subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -870,6 +872,34 @@ public void useTestNG(Action<? super TestFrameworkOptions> testFrameworkConfigur
         useTestFramework(new TestNGTestFramework(this, (DefaultTestFilter) getFilter(), getInstantiator(), getClassLoaderCache()), testFrameworkConfigure);
     }
 
+    /**
+     * Specifies that JUnit Platform should be used to execute the tests. <p> To configure JUnit Platform specific options, see {@link #useJUnitPlatform(groovy.lang.Closure)}.
+     */
+    public void useJUnitPlatform() {
```

Comment:
The wrinkle with that is that the JUnit Platform JARs are versioned as 1.0 not 5.0 like the Jupiter (JUnit 5 proper) JARs. I agree that useJUnit5 will seem more obvious at first glance, but not sure if it accurately describes the feature.

Reply:
We (the JUnit team) would like to be able to increment the major version number in the future. How about exposing both `useJUnit5()` and `useJUnitPlatform()`?

Reply:
I'd even vote for just `useJUnitPlatform()`.

IMHO, it's better to educate people about the "JUnit Platform" now rather than calling JUnit Platform support "JUnit 5", especially since test engines other than the JUnit Jupiter test engine can and will be run via `useJUnitPlatform()`.

Reply:
I'd agree with @sbrannen.

Reply:
Thanks for your guys' feedback. I'd probably not expose two different methods that do the same thing. Most of the time it just confuses users. We'll have to think about it.

Reply:
Excuse my ignorance, but why do we need such a method at all? Can't we just infer from the classpath?

Reply:
Yes, we can, but that will make things quite opaque in my opinion. We can sure infer `JUnit4`/`JUnit5`/`TestNG` from test runtime classpath, but why should we do this? With explicit `useJUnitPlatform`, build users can know without consulting documentation, "this build is using JUnit 5". 

What's more, we need some configuration closure like 

```
useJUnitPlatform {
     excludeTags 'tag1', 'tag2'
}
```

So I think an extra method is necessary here.

Reply:
To be clear, the reason I prefer this approach is that it is modeling how we test rather than _what we use to test_. In particular, I'd be interested in modeling, in the future, things like executing the same test suite on different target platforms.

Reply:
I think Bo is trying to follow the same approach that's already been taken for `useJUnit {...}` and `useTestNG {...}`. Whether or not this should be a separate plugin is probably beyond the scope of this PR. Whatever you decide, please don't call it `junit5` because there will be a JUnit 6 some day and it will hopefully also use the JUnit Platform, although predicting the future is always hard. 😉

Reply:
I'm happy with this too:
```
junitPlatform {
...
}
```